File: mycollab-services/src/main/java/com/mycollab/aspect/ClassInfoMap.java
package com.mycollab.aspect;

import java.util.HashMap;
import java.util.Map;

/**
 * @author MyCollab Ltd.
 * @since 5.0.1
 */
public class ClassInfoMap {
    private static Map<Class, ClassInfo> mapWrapper = new HashMap<>();

    public static void put(Class cls, ClassInfo classInfo) {
        mapWrapper.put(cls, classInfo);
    }

    public static ClassInfo getClassInfo(Class cls) {
        return mapWrapper.get(cls);
    }

    public static String getModule(Class cls) {
        return mapWrapper.get(cls).getModule();
    }

    public static String getType(Class cls) {
        return mapWrapper.get(cls).getType();
    }
}

File: mycollab-services/src/main/java/com/mycollab/aspect/ClassInfoMap.kt
package com.mycollab.aspect

import java.util.HashMap

/**
 * @author MyCollab Ltd.
 * @since 5.0.1
 */
object ClassInfoMap {
    private val mapWrapper = HashMap<Class<*>, ClassInfo>()

    fun put(cls: Class<*>, classInfo: ClassInfo) {
        mapWrapper.put(cls, classInfo)
    }

    fun getClassInfo(cls: Class<*>): ClassInfo? {
        return mapWrapper[cls]
    }

    fun getModule(cls: Class<*>): String {
        return mapWrapper[cls]!!.module!!
    }

    fun getType(cls: Class<*>): String {
        return mapWrapper[cls]!!.type!!
    }
}
